js/models/messages.js

```diff
@@ -1196,6 +1216,75 @@
       return errors[0][0];
     },
 
+    async handleMessageSentSuccess(sentMessage) {
+      const sentTo = this.get('sent_to') || [];
+
+      const isOurDevice = window.libsession.Protocols.MultiDeviceProtocol.isOurDevice(
+        sentMessage.device
+      );
+
+      // Handle the sync logic here
+      if (!isOurDevice && !this.get('synced') && !this.get('sentSync')) {
+        const contentDecoded = textsecure.protobuf.Content.decode(
+          sentMessage.plainTextBuffer
+        );
+        const { dataMessage } = contentDecoded;
+        this.sendSyncMessageOnly(dataMessage);
+
+        this.set({ sentSync: true });
+      } else if (isOurDevice && this.get('sentSync')) {
+        this.set({ synced: true });
+      }
+      const primaryPubKey = await libsession.Protocols.MultiDeviceProtocol.getPrimaryDevice(
+        sentMessage.device
+      );
+      this.set({
+        sent_to: _.union(sentTo, [primaryPubKey.key]),
+        sent: true,
+        expirationStartTimestamp: Date.now(),
+        // unidentifiedDeliveries: result.unidentifiedDeliveries,
+      });
+
+      await window.Signal.Data.saveMessage(this.attributes, {
+        Message: Whisper.Message,
+      });
+      this.getConversation().updateLastMessage();
+
+      this.trigger('sent', this);
+    },
+
+    async handleMessageSentFailure(sentMessage, error) {
+      if (error instanceof Error) {
+        this.saveErrors(error);
+        if (error.name === 'SignedPreKeyRotationError') {
+          await window.getAccountManager().rotateSignedPreKey();
+        } else if (error.name === 'OutgoingIdentityKeyError') {
+          const c = ConversationController.get(sentMessage.device);
+          await c.getProfiles();
+        }
+      }
+
+      const expirationStartTimestamp = Date.now();
+      if (
+        sentMessage.device === window.textsecure.storage.user.getNumber() &&
```

Comment:
This should check if the device we sent to is one of our devices

Reply:
fixed